Boost throws C++11 deprecation warnings

<!-- Please fill in the following information to report a problem with Cantera. If you have a question about using Cantera, please post it on our Google Users' Group (https://groups.google.com/forum/#!forum/cantera-users). Feature enhancements should be discussed in the dedicated Cantera enhancements repository (https://github.com/Cantera/enhancements/new/choose) -->

**Problem description**

<!-- A clear and concise description of what the bug is. -->

Compilation with recent versions of Boost produces deprecation warnings for C++11, for example:
```
g++ -o build/src/thermo/RedlichKwongMFTP.os -c -std=c++11 -O3 -Wno-inline -g -Wall -include src/pch/system.h -fPIC -DNDEBUG -Iinclude -Iinclude/cantera/ext -Ibuild/src -I/opt/homebrew/Caskroom/miniforge/base/envs/cantera-dev/include src/thermo/RedlichKwongMFTP.cpp

In file included from src/thermo/RedlichKwongMFTP.cpp:11:

In file included from /opt/homebrew/Caskroom/miniforge/base/envs/cantera-dev/include/boost/math/tools/roots.hpp:19:

/opt/homebrew/Caskroom/miniforge/base/envs/cantera-dev/include/boost/math/tools/config.hpp:23:6: warning: "The minimum language standard to use Boost.Math will be C++14 starting in July 2023 (Boost 1.82 release)" [-W#warnings]

#    warning "The minimum language standard to use Boost.Math will be C++14 starting in July 2023 (Boost 1.82 release)"

     ^

1 warning generated.
```

**Steps to reproduce**

<!-- A minimal, complete, and reproducible example demonstrating the problem. See https://stackoverflow.com/help/minimal-reproducible-example for additional suggestions on how to create such an example. -->

Runs `scons build` with recent boost 

**Behavior**

<!-- Describe the result of executing the above steps, and how this differs from what you expect to happen. -->

**System information**

- Cantera version: 3.0.0a2
- OS: macOS 12.5
- Python/MATLAB/other software versions:  boost-cpp 1.80.0 installed via conda / miniforge

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -76,8 +76,8 @@
 * Class names use `InitialCapsNames`
 * Methods use `camelCaseNames`
 * Do not indent the contents of namespaces
-* Code should follow the C++11 standard, with minimum required compiler versions
-  GCC 4.8, Clang 3.4, MSVC 14.0 (2015) and Intel 15.0.
+* Code should follow the C++14 standard, with minimum required compiler versions
+  GCC 6.1, Clang 3.4, MSVC 14.1 (2017) and Intel 17.0.
 * Avoid manual memory management (that is, `new` and `delete`), preferring to use
   standard library containers, as well as `std::unique_ptr` and
   `std::shared_ptr` when dynamic allocation is required.

--- a/SConstruct
+++ b/SConstruct
@@ -196,11 +196,11 @@ config_options = [
     Option(
         "cxx_flags",
         """Compiler flags passed to the C++ compiler only. Separate multiple
-           options with spaces, for example, "cxx_flags='-g -Wextra -O3 --std=c++11'"
+           options with spaces, for example, "cxx_flags='-g -Wextra -O3 --std=c++14'"
            """,
         {
             "cl": "/EHsc",
-            "default": "-std=c++11"
+            "default": "-std=c++14"
         }),
     Option(
         "CC",

--- a/SConstruct
+++ b/SConstruct
@@ -200,7 +200,6 @@ config_options = [
            """,
         {
             "cl": "/EHsc",
-            "Cygwin": "-std=gnu++11", # See http://stackoverflow.com/questions/18784112
             "default": "-std=c++11"
         }),
     Option(
@@ -751,7 +750,8 @@ env = Environment(tools=toolchain+["textfile", "subst", "recursiveInstall", "wix
 env["OS"] = platform.system()
 env["OS_BITS"] = int(platform.architecture()[0][:2])
 if "cygwin" in env["OS"].lower():
-    env["OS"] = "Cygwin" # remove Windows version suffix
+    logger.error(f"Error: Operating system {os.name!r} is no longer supported.")
+    sys.exit(1)
 
 if "FRAMEWORKS" not in env:
     env["FRAMEWORKS"] = []
@@ -791,8 +791,6 @@ if env["OS"] == "Darwin":
         config.select("apple-clang")
 
 if "gcc" in env.subst("$CC") or "gnu-cc" in env.subst("$CC"):
-    if env["OS"] == "Cygwin":
-        config.select("Cygwin")
     config.select("gcc")
 
 elif env["CC"] == "cl": # Visual Studio

--- a/interfaces/cython/SConscript
+++ b/interfaces/cython/SConscript
@@ -74,9 +74,6 @@ elif localenv['OS'] == 'Windows':
         # See https://github.com/python/cpython/pull/11283
         if py_version_full < parse_version("3.7.3"):
             localenv.Append(CPPDEFINES={"_hypot": "hypot"})
-elif localenv['OS'] == 'Cygwin':
-    # extract 'pythonX.Y' from 'libpythonX.Y.dll.a'
-    localenv.Append(LIBS=pylib[3:-6])
 
 if "numpy_1_7_API" in localenv:
     localenv.Append(CPPDEFINES="NPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION")

--- a/interfaces/python_sdist/setup.py
+++ b/interfaces/python_sdist/setup.py
@@ -146,7 +146,7 @@ def configure_build():
         create_config("SOLARIS", 0)
 
     if sys.platform != "win32":
-        extra_compile_flags = ["-std=c++11", "-g0"]
+        extra_compile_flags = ["-std=c++14", "-g0"]
         sundials_configh = {
             "SUNDIALS_USE_GENERIC_MATH": "#define SUNDIALS_USE_GENERIC_MATH 1",
             "SUNDIALS_BLAS_LAPACK": "/* #undef SUNDIALS_BLAS_LAPACK */"

--- a/site_scons/buildutils.py
+++ b/site_scons/buildutils.py
@@ -1263,10 +1263,3 @@ def get_pip_install_location(
         print(json.dumps(scheme))
     """)
     return json.loads(get_command_output(python_cmd, "-c", install_script))
-
-
-# Monkey patch for SCons Cygwin bug
-# See https://github.com/SCons/scons/issues/2664
-if "cygwin" in platform.system().lower():
-    import SCons.Node.FS
-    SCons.Node.FS._my_normcase = lambda x: x

--- a/test_problems/SConscript
+++ b/test_problems/SConscript
@@ -26,7 +26,7 @@ localenv['ENV']['CANTERA_DATA'] = (Dir('#build/data').abspath + os.pathsep +
                                    Dir('#test/data').abspath)
 
 # Add build/lib in order to find Cantera shared library
-if env['OS'] == 'Windows' or env['OS'] == 'Cygwin':
+if env["OS"] == "Windows":
     localenv.PrependENVPath('PATH', Dir('#build/lib').abspath)
 elif env['OS'] == 'Darwin':
     localenv.PrependENVPath('DYLD_LIBRARY_PATH', Dir('#build/lib').abspath)

